### PR TITLE
fix(session): harden stream removal ownership

### LINKED ISSUE #10835: Close the desktop process-store incarnation desync that bypasses the deletion claim
## Context

Follow-up from #10805 (fix(session): unify stream removal and metadata reads).

The desktop process-store deletion fallback (`packages/desktop/src/main/desktopProcessStores.ts`) keys its `hasStreamDeletionClaim` check on a locally-bumped incarnation counter whose bump gate (`pendingRemovals.has(streamId)`) is cleared when a removal settles — unlike `SessionState`'s re-claim gate, which keys on the persistent removal tombstone.

## Details

Trace (from the review thread): (1) stream S is deleted and the deletion commits → the tombstone persists, `pendingRemovals` is cleared. (2) The user re-runs the workflow → `claimStreamIdentity` bumps the state-side incarnation to 1, but the desktop counter stays 0. (3) The user deletes the re-claimed stream → the live `ProgressBackend` path registers a claim at incarnation 1, but the fallback checks `hasStreamDeletionClaim(streamId, 0)`, misses it, and launches a second, independently-guarded `deleteStreamAfterOwnedExecutionRelease` — the dual concurrent deletion the claim registry was added to prevent.

Fix shape (either): make `hasStreamDeletionClaim` incarnation-agnostic (a claim present at microtask time always means a live presentation owns the removal — claims are released on settle and on error), or align the process-side bump gate with the fact applier's tombstone-based gate, which requires exposing that state to the main process.

## References

- PR: #10805
- Review thread: https://github.com/LionSR/TeXRA/pull/10805#discussion_r3797157363
- Umbrella: #10834

### LINKED ISSUE #10837: Filter provisionally-removed streams in selectableStreamNames
## Context

Follow-up from #10805 (fix(session): unify stream removal and metadata reads).

## Details

`SessionState.selectableStreamNames()` iterates `streamLogs.keys()` directly rather than filtering `isStreamRemoved`, so a provisionally-removed stream (tombstoned but not yet durably deleted) can still appear in the rail for the narrow window between barrier installation and commit. Display-freshness nuance rather than a correctness issue in the fenced deletion path — kept on record as still open at merge by two review passes.

## References

- PR: #10805
- Review: https://github.com/LionSR/TeXRA/pull/10805#issuecomment-5317515374 (tenth pass, "Still open, not blocking")
- Umbrella: #10834

### PR DESCRIPTION
**Pull request by LionSR**
## Summary

Batch the related stream-removal follow-ups into one lifecycle change:

- hide streams from the selectable tab rail as soon as their removal barrier is installed
- let the desktop fallback recognize any live presentation deletion claim
- keep incarnation tracking in its authoritative owner, SessionState
- retire retained command- and fact-path tombstones before rebuilding presentation so failed cleanup restores the tab and active selection

## Root causes

The selectable stream list did not consult the removal barrier, leaving a provisionally removed tab visible while durable deletion was pending. Separately, the desktop fallback compared against stale process-local incarnation state and could race a live ProgressBackend owner. Combining the fixes exposed ordering issues on both command and session-fact retention paths: presentation rebuilt before the provisional tombstone retired, so a still-live stream remained filtered.

## Validation

- corepack pnpm typecheck:workspace
- corepack pnpm typecheck:test-kernel
- targeted four-file Vitest run: 168 passed
- targeted ESLint across all changed production and test files
- git diff --check

Fixes #10835.
Fixes #10837.

### CODE REVIEW

File: packages/desktop/src/main/desktopProcessStores.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export async function initializeDesktopProcessStores(session: SessionHandle) {
         // Check in the following microtask so the process fallback runs only
         // when no presentation owns the removal.
         queueMicrotask(() => {
-          if (stores.hasStreamDeletionClaim(streamId, expectedIncarnation)) {
+          if (stores.hasStreamDeletionClaim(streamId)) {
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/desktop/src/main/desktopProcessStores.ts:61]: Addressed in b5cf488184 with a desktop process-owner regression that reproduces the cross-incarnation race: the first presentation claim is observed and retired, a fresh multi-agent workflow identity and live handle re-claim the stream, then a second removal is held at the newer presentation claim. The test asserts the process fallback is never invoked and the transcript remains present while that claim is held. The combined four-file run now passes 167 tests.

File: src/controllers/progressView/backend/ProgressBackend.ts [hunk 2]
```diff
@@ -527,6 +527,19 @@ export class ProgressBackend {
...
+      // Retire a retained command-owned tombstone before rebuilding the tab
+      // rail. selectableStreamNames() deliberately hides provisional
+      // removals, so repairing presentation first would omit the stream that
+      // durable cleanup just reported as still live.
+      this.factApplier.completeCommandRemoval(
+        stream,
+        commandRemoval.incarnation,
+        retained,
+        commandRemoval.created,
+      );
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/controllers/progressView/backend/ProgressBackend.ts:541]: The retained-path regression is already caught at the durable desktop boundary by `DesktopAgentExecution.vitest.ts` test `resyncs a stream retained after durable cleanup fails` (around line 1944). When #10838 was first merged into this branch with the repair block still before `completeCommandRemoval`, that existing test failed exactly on this interaction: expected `activeStream: retained-stream`, received `activeStream: ""`. After moving tombstone retirement before rebuild, the same combined four-file run passed (now 167/167). That test drives the inbound delete command, forces `stageDeleteStream` to reject, and asserts both the active stream and retained content are resynchronized, so reverting this reorder fails it. I am not adding a second test for the same reproduced defect per the repository zero-duplicate-test discipline.

File: src/controllers/session/SessionState.ts [hunk 3]
```diff
@@ -193,6 +193,7 @@ export class SessionState {
   selectableStreamNames(): StreamTabId[] {
     return this.streamLogs
       .keys()
+      .filter((name) => !this.isStreamRemoved(name))
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/controllers/session/SessionState.ts:196]: Addressed in 8833664ac4. SessionFactApplier now passes its idempotent deletion-settlement callback into the host delete; ProgressBackend invokes it for active/failed fact-path outcomes before rebuilding the selectable rail. The normal promise settlement still handles deleted/superseded and hosts that do not support the early hook. Added `retires a retained fact removal before rebuilding the selectable rail`, which captures `selectableStreamNames()` inside the rebuild and asserts the retained stream is already visible. Workspace/test-kernel typechecks, targeted ESLint, and the combined four-file run are green (168 tests).

File: src/controllers/session/SessionFactApplier.ts [hunk 4]
```diff
@@ -80,6 +80,7 @@ export type SessionFactApplierOptions = {
   deleteStream: (
     stream: StreamTabId,
     expectedIncarnation?: number,
+    beforeRetainedRepair?: (outcome: 'active' | 'failed') => void,
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/controllers/session/SessionFactApplier.ts:91]: Addressed in 1ad081ff1d. The SessionFactApplierOptions contract now states that hosts with a retained presentation-repair path must invoke the hook with active/failed before enumerating selectable streams, explains that the callback retires the provisional barrier and replays buffered facts, and explicitly permits hosts without such a repair path to rely on normal promise settlement.

File: src/controllers/session/SessionState.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (LionSR) [reply src/controllers/session/SessionState.ts:196]: Addressed in 203c420472: the source comment now states that provisional removals are excluded before creation-time ordering and rotation.

File: src/controllers/progressView/backend/ProgressBackend.ts [hunk 5]
```diff
@@ -527,6 +528,24 @@ export class ProgressBackend {
     }
     releaseDeletionClaim();
 
+    if (commandRemoval) {
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/controllers/progressView/backend/ProgressBackend.ts:531]: Addressed in 437314c3b4: the existing retained command-deletion test now captures selectableStreamNames() during rebuild and asserts both the retained rail membership and retired provisional-removal state.

File: src/controllers/session/SessionFactApplier.ts [hunk 6]
```diff
@@ -76,10 +76,19 @@ export type SessionFactApplierOptions = {
...
    * the host can refuse to delete a deterministic identity that a fresh run
    * re-claimed while the deletion was queued.
+   *
+   * A host that rebuilds retained presentation state must invoke
+   * `beforeRetainedRepair` with its `active`/`failed` outcome before that
+   * rebuild enumerates selectable streams. The callback retires the
+   * provisional removal barrier and replays buffered facts; rebuilding first
+   * would omit the still-live stream because selectable projections hide
+   * provisional removals. Hosts without a selectable-stream repair path may
+   * ignore the optional hook and let normal promise settlement retire it.
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/controllers/session/SessionFactApplier.ts:86]: Addressed in 5590ea823a on the still-open batched #10833 branch after #10839 merged. The comment now limits normal-settlement retirement to retained active/failed outcomes, matching the void-host contract.
